src/AggregateFunctions/AggregateFunctionWelchTTest.h

```diff
@@ -0,0 +1,315 @@
+        sum_y += other.sum_y;
+        size_x += other.size_x;
+        size_y += other.size_y;
+        mean_x = static_cast<Float64>(sum_x) / size_x;
+        mean_y = static_cast<Float64>(sum_y) / size_y;
+        square_sum_x += other.square_sum_x;
+        square_sum_y += other.square_sum_y;
+    }
+
+    void serialize(WriteBuffer &buf) const
+    {
+        writeBinary(mean_x, buf);
+        writeBinary(mean_y, buf);
+        writeBinary(sum_x, buf);
+        writeBinary(sum_y, buf);
+        writeBinary(square_sum_x, buf);
+        writeBinary(square_sum_y, buf);
+        writeBinary(size_x, buf);
+        writeBinary(size_y, buf);
+    }
+
+    void deserialize(ReadBuffer &buf)
+    {
+        readBinary(mean_x, buf);
+        readBinary(mean_y, buf);
+        readBinary(sum_x, buf);
+        readBinary(sum_y, buf);
+        readBinary(square_sum_x, buf);
+        readBinary(square_sum_y, buf);
+        readBinary(size_x, buf);
+        readBinary(size_y, buf);
+    }
+
+    size_t get_size_y() const
+    {
+        return size_y;
+    }
+
+    size_t get_size_x() const
+    {
+        return size_x;
+    }
+
+    Float64 get_sx() const
+    {
+        return static_cast<Float64>(square_sum_x + size_x * mean_x * mean_x - 2 * mean_x * sum_x) / (size_x - 1);
+    }
+
+    Float64 get_sy() const
+    {
+        return static_cast<Float64>(square_sum_y + size_y * mean_y * mean_y - 2 * mean_y * sum_y) / (size_y - 1);
+    }
+
+    Float64 get_T(Float64 sx, Float64 sy) const
+    {
+        if (sx == 0 && sy == 0)
+        {
+            throw Exception("division by zero encountered in Aggregate function WelchTTest", ErrorCodes::BAD_ARGUMENTS);
+        }
+
+        if (sx == -sy && size_x == size_y)
+        {
+            throw Exception("division by zero encountered in Aggregate function WelchTTest", ErrorCodes::BAD_ARGUMENTS);
+        }
+
+        return static_cast<Float64>(mean_x - mean_y) / std::sqrt(sx / size_x + sy / size_y);
+    }
+
+    Float64 get_degrees_of_freed(Float64 sx, Float64 sy) const
+    {
+        return static_cast<Float64>(sx / size_x + sy / size_y) * (sx / size_x + sy / size_y) /
+               ((sx * sx / (size_x * size_x * (size_x - 1))) + (sy * sy / (size_y * size_y * (size_y - 1))));
+    }
+
+    UInt8 get_result(Float64 t, Float64 dof, Float64 parametr) const
+    {
+        //find our table
+        int table = 0;
+        for (int i = 0; i < SIGN_LVL_CNT; ++i)
+        {
+            if (CriticalValuesTable[i][0] == parametr)
+            {
+                table = i;
+            }
+        }
+
+        //round or make infinity dof
+        int i_dof = static_cast<int>(dof);
+
+        if (i_dof > 100)
+        {
+            i_dof = 101;
+        }
+
+        if (i_dof < 100)
+        {
+            i_dof = 1;
+        }
+
+        //check if abs of t is greater than table[dof]
+        t = abs(t);
+        if (t >= CriticalValuesTable[table][i_dof])
+        {
+            return static_cast<UInt8>(1);
```

Comment:
As I understood, Welch test's null hypotesis is "Two population means are equal". So, I think it's more intutive to return 1 if hypotesis was not rejected and populations has equal means. Also it will be usefull to have simililar functions, that return T statistic and p-value, not only bool.